File: plugin/src/main/java/com/typelead/intellij/plugin/eta/lang/lexer/EtaSyntaxHighlightingLexer.java
package com.typelead.intellij.plugin.eta.lang.lexer;

import com.intellij.lexer.FlexAdapter;

public class EtaSyntaxHighlightingLexer extends FlexAdapter {
  public EtaSyntaxHighlightingLexer() {
    super(new _EtaSyntaxHighlightingLexer());
  }
}

File: plugin/src/main/eta/FFI/Com/TypeLead/IntelliJ/Plugin/Eta/Lang/Lexer/EtaSyntaxHighlightingLexer.hs
module FFI.Com.TypeLead.IntelliJ.Plugin.Eta.Lang.Lexer.EtaSyntaxHighlightingLexer where

import P

import FFI.Com.IntelliJ.Lexer.Lexer

data EtaSyntaxHighlightingLexer = EtaSyntaxHighlightingLexer
  @com.typelead.intellij.plugin.eta.lang.lexer.EtaSyntaxHighlightingLexer
  deriving Class

type instance Inherits EtaSyntaxHighlightingLexer = '[Object, Lexer]

foreign import java unsafe "@new" newEtaSyntaxHighlightingLexer :: Java a EtaSyntaxHighlightingLexer
